kv/kvnemesis: TestKVNemesisMultiNode failed

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2734059&tab=buildLog) on [master@1b18fc1fa8d9194d5e7f9aaa857425d80f0e4970](https://github.com/cockroachdb/cockroach/commits/1b18fc1fa8d9194d5e7f9aaa857425d80f0e4970):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode786161560
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:85: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"b81ab7eb", 2) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:618
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:353
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:100
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:81
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"b81ab7eb", 2) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:87: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode786161560
--- FAIL: TestKVNemesisMultiNode (0.60s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2739197&tab=buildLog) on [master@6de4313ec216161c79fe725fcc31fc87ef1804ea](https://github.com/cockroachdb/cockroach/commits/6de4313ec216161c79fe725fcc31fc87ef1804ea):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode579305741
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:85: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"e6c3fa5e", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1614755980.855514634,0 epo=1 pro=1614755980.855325030,0 with repl=(n1,s1):1 seq=0 start=1614755980.886729612,0 exp=<nil> pro=1614755980.886729612,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n4,s4):2 seq=2 start=1614755980.855514634,0 epo=1 pro=1614755980.855325030,0 with repl=(n1,s1):1 seq=0 start=1614755980.886729612,0 exp=<nil> pro=1614755980.886729612,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:100
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:81
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"e6c3fa5e", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1614755980.855514634,0 epo=1 pro=1614755980.855325030,0 with repl=(n1,s1):1 seq=0 start=1614755980.886729612,0 exp=<nil> pro=1614755980.886729612,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n4,s4):2 seq=2 start=1614755980.855514634,0 epo=1 pro=1614755980.855325030,0 with repl=(n1,s1):1 seq=0 start=1614755980.886729612,0 exp=<nil> pro=1614755980.886729612,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:87: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode579305741
--- FAIL: TestKVNemesisMultiNode (0.77s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2743535&tab=buildLog) on [master@37cdf01299861235888af3411e699480f5a2d670](https://github.com/cockroachdb/cockroach/commits/37cdf01299861235888af3411e699480f5a2d670):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode198946516
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:85: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"e93d9dc4", 3) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:100
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:81
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"e93d9dc4", 3) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:87: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode198946516
--- FAIL: TestKVNemesisMultiNode (0.63s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2747703&tab=buildLog) on [master@b703e663da8ededaee2e28fc39a24e3880ae54cf](https://github.com/cockroachdb/cockroach/commits/b703e663da8ededaee2e28fc39a24e3880ae54cf):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode387236614
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:85: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"2085254b", 3) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:100
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:81
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"2085254b", 3) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:87: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode387236614
--- FAIL: TestKVNemesisMultiNode (0.64s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2753372&tab=buildLog) on [master@a69e6549a71f5a0e83eb13509001f4d7351050fb](https://github.com/cockroachdb/cockroach/commits/a69e6549a71f5a0e83eb13509001f4d7351050fb):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode049091563
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"3b23d7fd", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1615100868.961781083,0 epo=1 pro=1615100868.961420487,0 with repl=(n1,s1):1 seq=0 start=1615100868.999518092,0 exp=<nil> pro=1615100868.999518092,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n4,s4):2 seq=2 start=1615100868.961781083,0 epo=1 pro=1615100868.961420487,0 with repl=(n1,s1):1 seq=0 start=1615100868.999518092,0 exp=<nil> pro=1615100868.999518092,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"3b23d7fd", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1615100868.961781083,0 epo=1 pro=1615100868.961420487,0 with repl=(n1,s1):1 seq=0 start=1615100868.999518092,0 exp=<nil> pro=1615100868.999518092,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n4,s4):2 seq=2 start=1615100868.961781083,0 epo=1 pro=1615100868.961420487,0 with repl=(n1,s1):1 seq=0 start=1615100868.999518092,0 exp=<nil> pro=1615100868.999518092,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode049091563
--- FAIL: TestKVNemesisMultiNode (1.17s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2757787&tab=buildLog) on [master@93c38e97ddb9d18721eed850180c103fe3c45547](https://github.com/cockroachdb/cockroach/commits/93c38e97ddb9d18721eed850180c103fe3c45547):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode842178868
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"a4d28c77", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1615276193.079342606,0 epo=1 pro=1615276193.078941519,0 with repl=(n1,s1):1 seq=0 start=1615276193.081319832,0 exp=<nil> pro=1615276193.081319832,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1615276193.079342606,0 epo=1 pro=1615276193.078941519,0 with repl=(n1,s1):1 seq=0 start=1615276193.081319832,0 exp=<nil> pro=1615276193.081319832,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"a4d28c77", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1615276193.079342606,0 epo=1 pro=1615276193.078941519,0 with repl=(n1,s1):1 seq=0 start=1615276193.081319832,0 exp=<nil> pro=1615276193.081319832,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1615276193.079342606,0 epo=1 pro=1615276193.078941519,0 with repl=(n1,s1):1 seq=0 start=1615276193.081319832,0 exp=<nil> pro=1615276193.081319832,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode842178868
--- FAIL: TestKVNemesisMultiNode (5.36s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2766073&tab=buildLog) on [master@fc68d927f946103a757888fcbd3ac20904c57a6b](https://github.com/cockroachdb/cockroach/commits/fc68d927f946103a757888fcbd3ac20904c57a6b):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode728807589
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"47c09f46", 4) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"47c09f46", 4) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode728807589
--- FAIL: TestKVNemesisMultiNode (1.15s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2775521&tab=buildLog) on [master@bdff5338ca725bf1cfddf7e3f648bbf02ab42999](https://github.com/cockroachdb/cockroach/commits/bdff5338ca725bf1cfddf7e3f648bbf02ab42999):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode003609470
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"5b4746aa", 4) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"5b4746aa", 4) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode003609470
--- FAIL: TestKVNemesisMultiNode (1.13s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2776993&tab=buildLog) on [master@f3c8f6ed74494f93bd1eb01b1d7569127f127906](https://github.com/cockroachdb/cockroach/commits/f3c8f6ed74494f93bd1eb01b1d7569127f127906):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode396686203
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"148a0682", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1615790437.893944621,0 epo=1 pro=1615790437.893016762,0 with repl=(n1,s1):1 seq=0 start=1615790437.916012208,0 exp=<nil> pro=1615790437.916012208,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1615790437.893944621,0 epo=1 pro=1615790437.893016762,0 with repl=(n1,s1):1 seq=0 start=1615790437.916012208,0 exp=<nil> pro=1615790437.916012208,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"148a0682", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1615790437.893944621,0 epo=1 pro=1615790437.893016762,0 with repl=(n1,s1):1 seq=0 start=1615790437.916012208,0 exp=<nil> pro=1615790437.916012208,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1615790437.893944621,0 epo=1 pro=1615790437.893016762,0 with repl=(n1,s1):1 seq=0 start=1615790437.916012208,0 exp=<nil> pro=1615790437.916012208,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode396686203
--- FAIL: TestKVNemesisMultiNode (1.33s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2796111&tab=buildLog) on [master@3d19b2cf6b290a152b23722fc32e995eed3b437b](https://github.com/cockroachdb/cockroach/commits/3d19b2cf6b290a152b23722fc32e995eed3b437b):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode775967295
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"48ca8b46", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1616221348.715285078,0 epo=1 pro=1616221348.715072682,0 with repl=(n1,s1):1 seq=0 start=1616221348.887276887,0 exp=<nil> pro=1616221348.887276887,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1616221348.715285078,0 epo=1 pro=1616221348.715072682,0 with repl=(n1,s1):1 seq=0 start=1616221348.887276887,0 exp=<nil> pro=1616221348.887276887,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"48ca8b46", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1616221348.715285078,0 epo=1 pro=1616221348.715072682,0 with repl=(n1,s1):1 seq=0 start=1616221348.887276887,0 exp=<nil> pro=1616221348.887276887,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1616221348.715285078,0 epo=1 pro=1616221348.715072682,0 with repl=(n1,s1):1 seq=0 start=1616221348.887276887,0 exp=<nil> pro=1616221348.887276887,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode775967295
--- FAIL: TestKVNemesisMultiNode (5.16s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2797684&tab=buildLog) on [master@893643b63ea0b1cfa4888c6b73b5c68a9c100c3a](https://github.com/cockroachdb/cockroach/commits/893643b63ea0b1cfa4888c6b73b5c68a9c100c3a):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode111655860
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"b6d5dab1", 2) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"b6d5dab1", 2) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode111655860
--- FAIL: TestKVNemesisMultiNode (1.12s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2803357&tab=buildLog) on [master@3f9f18ebf27dc0fdbf7fa0fce639be1513c12623](https://github.com/cockroachdb/cockroach/commits/3f9f18ebf27dc0fdbf7fa0fce639be1513c12623):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode295798159
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"d64828a0", 3) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"d64828a0", 3) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:505
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode295798159
--- FAIL: TestKVNemesisMultiNode (1.15s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2823408&tab=buildLog) on [master@d891594d3c998f153b88f631e3c89ac7d12c2a6e](https://github.com/cockroachdb/cockroach/commits/d891594d3c998f153b88f631e3c89ac7d12c2a6e):

```
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"84745278", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.899911531,0 exp=<nil> pro=1617000610.899911531,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.899911531,0 exp=<nil> pro=1617000610.899911531,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"6ff672f7", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.905154847,0 exp=<nil> pro=1617000610.905154847,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.905154847,0 exp=<nil> pro=1617000610.905154847,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"6ff672f7", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.905154847,0 exp=<nil> pro=1617000610.905154847,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.905154847,0 exp=<nil> pro=1617000610.905154847,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"723c45b3", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.929346329,0 exp=<nil> pro=1617000610.929346329,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.929346329,0 exp=<nil> pro=1617000610.929346329,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"723c45b3", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.929346329,0 exp=<nil> pro=1617000610.929346329,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1617000610.874884348,0 epo=1 pro=1617000610.874670825,0 with repl=(n1,s1):1 seq=0 start=1617000610.929346329,0 exp=<nil> pro=1617000610.929346329,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode967354694
--- FAIL: TestKVNemesisMultiNode (4.96s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2842093&tab=buildLog) on [master@9891b8db5ea2e9339a04fd2731a16658103e969f](https://github.com/cockroachdb/cockroach/commits/9891b8db5ea2e9339a04fd2731a16658103e969f):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode010515030
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"c63f363f", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617344475.331209739,0 epo=1 pro=1617344475.330256239,0 with repl=(n1,s1):1 seq=0 start=1617344475.476365348,0 exp=<nil> pro=1617344475.476365348,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n3,s3):2 seq=2 start=1617344475.331209739,0 epo=1 pro=1617344475.330256239,0 with repl=(n1,s1):1 seq=0 start=1617344475.476365348,0 exp=<nil> pro=1617344475.476365348,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"c63f363f", 1) // cannot replace lease repl=(n3,s3):2 seq=2 start=1617344475.331209739,0 epo=1 pro=1617344475.330256239,0 with repl=(n1,s1):1 seq=0 start=1617344475.476365348,0 exp=<nil> pro=1617344475.476365348,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n3,s3):2 seq=2 start=1617344475.331209739,0 epo=1 pro=1617344475.330256239,0 with repl=(n1,s1):1 seq=0 start=1617344475.476365348,0 exp=<nil> pro=1617344475.476365348,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode010515030
--- FAIL: TestKVNemesisMultiNode (5.57s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2845289&tab=buildLog) on [master@df07592e26735be7362ef2b815ee163ead00739e](https://github.com/cockroachdb/cockroach/commits/df07592e26735be7362ef2b815ee163ead00739e):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode059915257
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"7c52eb95", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.628373299,0 exp=<nil> pro=1617432664.628373299,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.628373299,0 exp=<nil> pro=1617432664.628373299,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"7c52eb95", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.628373299,0 exp=<nil> pro=1617432664.628373299,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.628373299,0 exp=<nil> pro=1617432664.628373299,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"6625d470", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.701637883,0 exp=<nil> pro=1617432664.701637883,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.701637883,0 exp=<nil> pro=1617432664.701637883,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"6625d470", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.701637883,0 exp=<nil> pro=1617432664.701637883,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n4,s4):2 seq=2 start=1617432664.579909696,0 epo=1 pro=1617432664.579689746,0 with repl=(n1,s1):1 seq=0 start=1617432664.701637883,0 exp=<nil> pro=1617432664.701637883,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode059915257
--- FAIL: TestKVNemesisMultiNode (1.15s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2855389&tab=buildLog) on [master@651184b3cfbd7c6d336bb49e7152e3d48ea06416](https://github.com/cockroachdb/cockroach/commits/651184b3cfbd7c6d336bb49e7152e3d48ea06416):

```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode942691456
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"724d8a2a", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617777779.042326013,0 epo=1 pro=1617777779.041851333,0 with repl=(n1,s1):1 seq=0 start=1617777779.231175602,0 exp=<nil> pro=1617777779.231175602,0: couldn't request lease for (n1,s1):1: liveness record not found in cache: cannot replace lease repl=(n4,s4):2 seq=2 start=1617777779.042326013,0 epo=1 pro=1617777779.041851333,0 with repl=(n1,s1):1 seq=0 start=1617777779.231175602,0 exp=<nil> pro=1617777779.231175602,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"724d8a2a", 1) // cannot replace lease repl=(n4,s4):2 seq=2 start=1617777779.042326013,0 epo=1 pro=1617777779.041851333,0 with repl=(n1,s1):1 seq=0 start=1617777779.231175602,0 exp=<nil> pro=1617777779.231175602,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Wraps: (3) cannot replace lease repl=(n4,s4):2 seq=2 start=1617777779.042326013,0 epo=1 pro=1617777779.041851333,0 with repl=(n1,s1):1 seq=0 start=1617777779.231175602,0 exp=<nil> pro=1617777779.231175602,0: couldn't request lease for (n1,s1):1: liveness record not found in cache
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.LeaseRejectedError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode942691456
--- FAIL: TestKVNemesisMultiNode (5.14s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[kv/kvnemesis.TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2860257&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2860257&tab=artifacts#/) on [master @ 2c50690c86b0de434946bb74f0e63aea1907ea20](https://github.com/cockroachdb/cockroach/commits/2c50690c86b0de434946bb74f0e63aea1907ea20):


```
=== RUN   TestKVNemesisMultiNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode848721057
    test_log_scope.go:74: use -show-logs to present logs inline
    kvnemesis_test.go:92: failure:
        error applying x.TransferLeaseOperation(ctx, /Table/50/"95ea6651", 4) // replica not found in RangeDescriptor: replica not found in RangeDescriptor
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:621
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:356
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/validator.go:74
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:104
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis_test.go:88
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (2) error applying x.TransferLeaseOperation(ctx, /Table/50/"95ea6651", 4) // replica not found in RangeDescriptor
        Wraps: (3)
          | (opaque error wrapper)
          | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          | reportable 0:
          |
          | github.com/cockroachdb/cockroach/pkg/roachpb.init
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/metadata_replicas.go:493
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5652
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.doInit
          | 	/usr/local/go/src/runtime/proc.go:5647
          | runtime.main
          | 	/usr/local/go/src/runtime/proc.go:191
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (4) replica not found in RangeDescriptor
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.leafError
    kvnemesis_test.go:94: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode848721057
--- FAIL: TestKVNemesisMultiNode (1.12s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -326,6 +326,8 @@ func (v *validator) processOp(txnID *string, op Operation) {
 			// Probably should be transparently retried.
 		} else if resultIsErrorStr(t.Result, `merge failed: range missing intent on its local descriptor`) {
 			// Probably should be transparently retried.
+		} else if resultIsErrorStr(t.Result, `merge failed: RHS range bounds do not match`) {
+			// Probably should be transparently retried.
 		} else {
 			v.failIfError(op, t.Result)
 		}

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -345,6 +345,10 @@ func (v *validator) processOp(txnID *string, op Operation) {
 			// Only VOTER_FULL replicas can currently hold a range lease.
 			// Attempts to transfer to lease to any other replica type are
 			// rejected.
+		} else if resultIsErrorStr(t.Result, `replica not found in RangeDescriptor`) {
+			// Only replicas that are part of the range can be given
+			// the lease. This case is hit if a TransferLease op races
+			// with a ChangeReplicas op.
 		} else if resultIsErrorStr(t.Result, `unable to find store \d+ in range`) {
 			// A lease transfer that races with a replica removal may find that
 			// the store it was targeting is no longer part of the range.
@@ -354,6 +358,10 @@ func (v *validator) processOp(txnID *string, op Operation) {
 		} else if resultIsError(t.Result, liveness.ErrRecordCacheMiss) {
 			// If the existing leaseholder has not yet heard about the transfer
 			// target's liveness record through gossip, it will return an error.
+		} else if resultIsErrorStr(t.Result, liveness.ErrRecordCacheMiss.Error()) {
+			// Same as above, but matches cases where ErrRecordCacheMiss is
+			// passed through a LeaseRejectedError. This is necessary until
+			// LeaseRejectedErrors works with errors.Cause.
 		} else {
 			v.failIfError(op, t.Result)
 		}

--- a/pkg/kv/kvserver/batcheval/cmd_subsume.go
+++ b/pkg/kv/kvserver/batcheval/cmd_subsume.go
@@ -79,7 +79,7 @@ func Subsume(
 	desc := cArgs.EvalCtx.Desc()
 	if !bytes.Equal(desc.StartKey, args.RightDesc.StartKey) ||
 		!bytes.Equal(desc.EndKey, args.RightDesc.EndKey) {
-		return result.Result{}, errors.AssertionFailedf("RHS range bounds do not match: %s != %s",
+		return result.Result{}, errors.Errorf("RHS range bounds do not match: %s != %s",
 			args.RightDesc, desc)
 	}
 
@@ -87,7 +87,7 @@ func Subsume(
 	// of operations in the AdminMerge transaction should make it impossible for
 	// these ranges to be nonadjacent, but double check.
 	if !bytes.Equal(args.LeftDesc.EndKey, desc.StartKey) {
-		return result.Result{}, errors.AssertionFailedf("ranges are not adjacent: %s != %s",
+		return result.Result{}, errors.Errorf("ranges are not adjacent: %s != %s",
 			args.LeftDesc.EndKey, desc.StartKey)
 	}
 
@@ -101,13 +101,13 @@ func Subsume(
 	if err != nil {
 		return result.Result{}, errors.Errorf("fetching local range descriptor: %s", err)
 	} else if intent == nil {
-		return result.Result{}, errors.AssertionFailedf("range missing intent on its local descriptor")
+		return result.Result{}, errors.Errorf("range missing intent on its local descriptor")
 	}
 	val, _, err := storage.MVCCGetAsTxn(ctx, readWriter, descKey, intent.Txn.WriteTimestamp, intent.Txn)
 	if err != nil {
 		return result.Result{}, errors.Errorf("fetching local range descriptor as txn: %s", err)
 	} else if val != nil {
-		return result.Result{}, errors.AssertionFailedf("non-deletion intent on local range descriptor")
+		return result.Result{}, errors.Errorf("non-deletion intent on local range descriptor")
 	}
 
 	// NOTE: the deletion intent on the range's meta2 descriptor is just as

--- a/pkg/kv/kvserver/replica_command.go
+++ b/pkg/kv/kvserver/replica_command.go
@@ -2327,7 +2327,8 @@ func (r *Replica) sendSnapshot(
 
 	snap, err := r.GetSnapshot(ctx, snapType, recipient.StoreID)
 	if err != nil {
-		return errors.Wrapf(err, "%s: failed to generate %s snapshot", r, snapType)
+		err = errors.Wrapf(err, "%s: failed to generate %s snapshot", r, snapType)
+		return errors.Mark(err, errMarkSnapshotError)
 	}
 	defer snap.Close()
 	log.Event(ctx, "generated snapshot")

